### ISSUE TITLE
Classes, Races and Backgrounds are not selectable

### ISSUE THREAD

**Issue by RonaldoBrum** [HeroCreationLab/hero-creation-tool#130]
Hi, I recently updated my foundry to version 10.288, using version 2.0.3 of dnd 5e. When I go to character building, the class, race and backgrounds are not selectable, just like in the attribute scrolling part, it doesn't show the button title (although if I click on that empty space it scrolls just right).
When consulting the log to see if the problem was discovered, I saw that error too (present in the prints)
![Screenshot_1](https://user-images.githubusercontent.com/60355256/200027321-e0e0e4ef-b3b8-440d-b6f6-5cb4fe4a8767.png)
![Screenshot_2](https://user-images.githubusercontent.com/60355256/200027327-1cfefa01-69c5-4656-8b1b-97dde62e5ff4.png)
![Screenshot_3](https://user-images.githubusercontent.com/60355256/200027330-c5186d41-f519-4019-9481-3bbcd756e290.png)
![Screenshot_4](https://user-images.githubusercontent.com/60355256/200027335-ee80fe50-2f92-4bf5-8b5b-60d74e3c6512.png)
![Screenshot_5](https://user-images.githubusercontent.com/60355256/200027338-db1e51e7-8934-4a21-8649-d9d588809996.png)

**Comment by ccjmk** [comment]
Hi! I will be looking at this as soon as possible. Right now im in the middle of moving so after I can get settled I will try to see what's wrong here.

**Comment by RonaldoBrum** [comment]
It's ok! I hope everything goes well with your move ^^

**Comment by ccjmk** [comment]
Hi! sorry for the long delay! I just did a quick smoke test with foundry v10.291, 5e 2.0.3 and module 1.9,0 and it seems to work. I was honestly too busy before to give proper attention to the screenshot you sent me, but that seems to indicate you have something that should be a spell that has no level. Do you have imported stuff from DDBImporter by chance?  I recall a folk that had a similar issue that got resolved by re-importing stuff. Could you try that please?

**Comment by RonaldoBrum** [comment]
No friend, I've been studying the problem and it seems to be the same as other issues posted here recently. It's like the creation tool can't read the compendiums which I have the classes, subs, races etc...

**Comment by RonaldoBrum** [comment]
I saw someone talk somewhere about something in the foundry having changed in current versions about what was previously read as an entity now being read as "type"

**Comment by RonaldoBrum** [comment]
I don't use ddb importer, everything in the compendiums that are not being read were created by me in foundry and in previous versions they were read. In updating my foundry they stopped being read.
I've already tried to run the hero creation tool alone and the problem continues.

**Comment by ccjmk** [comment]
by any chance, do you have  multiple types of stuff in the same compendium? like a single compendium with "my custom stuff" that has spells, classes, races, etc? those should go separately, because it sounds like this might be another possible reason!

**Comment by RonaldoBrum** [comment]
no, i'm extremely boring with my organizations, the races and classes i use were all created by me from scratch to be read by hero creation. Because I translate the tool to pt br.
And as I said, it's a problem that arose in the update. Something changed in the code or the way compendiums are read
![Screenshot_1](https://user-images.githubusercontent.com/60355256/211401676-161de86a-9a0a-4118-9797-dadb82fbef63.png)
![Screenshot_2](https://user-images.githubusercontent.com/60355256/211401685-f348918c-aa36-4f05-87b4-77a8cc54e1c6.png)
![Screenshot_3](https://user-images.githubusercontent.com/60355256/211401687-c387ac2f-03c8-414b-8745-7977664c05cc.png)
![Screenshot_4](https://user-images.githubusercontent.com/60355256/211401692-f5d5b60b-6ec0-486a-95e8-b7f45996c477.png)
![Screenshot_5](https://user-images.githubusercontent.com/60355256/211401695-9df7a534-916c-4fdc-9dfb-269696437cc7.png)
![Screenshot_6](https://user-images.githubusercontent.com/60355256/211401699-6bb3767c-8b54-4716-92a7-5f6ce31bf79b.png)
![Screenshot_7](https://user-images.githubusercontent.com/60355256/211401702-6f03cd15-4e9e-47ba-b2aa-f6fd38b7c8a9.png)
![Screenshot_8](https://user-images.githubusercontent.com/60355256/211401707-51cfb6af-8328-4f8f-a6a6-3678b5fc496d.png)

**Comment by RonaldoBrum** [comment]
As you can see in the prints, it stops loading because it has a problem rendering the race part in the tool. the background part seems to be working normally. The class part doesn't show the skills, maybe because they now have an advancement tab that the tool still doesn't read.

**Comment by ccjmk** [comment]
Ooooh I think I have a clue! but... it's not good if I land on the issue hahah if you have this  compendia from a long time ago, they might be structured on the old way. For example, before, the value `requirements` was under... `item.data.data.requirements` iirc, now its `item.system.requirements` or something like that. Let me see if I can find / make some code to check and update the compendiums

**Comment by RonaldoBrum** [comment]
yes I've had them for a long time, I thought it was something structural, because when dragging to the description of some skill it generated a link @name.world.data (or something like that) and now it generates a UUCC.name.world. date then I realized that there were structural changes by the new att

**Comment by ccjmk** [comment]
Could you send me the files of your compendiums?  I think sadly there's no easy workaround for this, but I can try to make some migration code for it. No promises though, I need to find time for this sadly :(

**Comment by Ancyker** [comment]
This happens when the `system` field is missing. It also happens when the `level` field is missing from spells.

**Comment by Ancyker** [comment]
> Could you send me the files of your compendiums? I think sadly there's no easy workaround for this, but I can try to make some migration code for it. No promises though, I need to find time for this sadly :(

This project currently has this issue: https://github.com/btbias/5e-complete

Note: I've put a bug report in with them to fix it. I feel it's really a bug in the compendium as this info _should_ be present, but it's probably also worth handling the missing fields in the module.

**Comment by d34d-p1x3l** [comment]
Any word with how this bug is going? I have the same issue. I guess it's only applicable to custom compendia. I spent a couple months putting all the info from PHB-DMG-Xanathar's-Tasha's-SwordCoast and others into compendia so my players would have all the same options in game as they would when we played in person. The compendia were originally just copies of the Foundry SRDs so I didn't think the formatting would be different. In response to Ancyker's comments, where are the system and level fields supposed to be in the compendia? They can be edited in notepad.

**Comment by cyn-louwho** [comment]
Hi! I am also having this issue with Foundry v11. 
I understand I am responding to a really old thread, but hopefully I can get some assistance. I am trying to input an Artificer class. It works fine until I start adding my features into the level up section. I followed the youtube tutorial, but as soon as I try to do anything with the features it causes all of the usual selections on the left to disappear. Thank you!

![image](https://github.com/HeroCreationLab/hero-creation-tool/assets/135558578/db61921c-b7e0-4ecb-a843-e8fc8a1b62c2)
![image](https://github.com/HeroCreationLab/hero-creation-tool/assets/135558578/d64db42f-04ee-4ddf-badd-6238839a28b6)
![image](https://github.com/HeroCreationLab/hero-creation-tool/assets/135558578/e17fca4f-3ca6-48b7-8e0c-6fe4447b63c0)
![image](https://github.com/HeroCreationLab/hero-creation-tool/assets/135558578/b4ebcd93-92d9-4022-b20e-8427d80cf991)
![image](https://github.com/HeroCreationLab/hero-creation-tool/assets/135558578/b19bf5c8-c9d5-48fa-b187-33b77ab2b290)
![image](https://github.com/HeroCreationLab/hero-creation-tool/assets/135558578/4659d0dd-5dee-4649-a73b-a29689ed3607)

**Comment by ccjmk** [comment]
Sorry for the long delay @d34d-p1x3l , it seems I missed your comment. If you have the compendia made a long time ago, they problably don't have the latest structure. Are they working on your classes if you add them manually (without my module)? I imagine we could get some help in the dnd5e channels to make a small script that would update those for you.
